Fixes daze absorb and duplicate target flags

RidingKeys opened the pull request:
Fixes Daze effect applying on shielded tagets that absorb the damage. Also fixes a target type that had a broken function that was identical to another(Fixes Nightlash)

Fixes these two bugs for sure(possibly some others, but unknown at this time.)
https://www.getmangos.eu/bug-tracker/mangos-zero/Daze-state-is-applied-incorrec-r752/

https://www.getmangos.eu/bug-tracker/mangos-zero/Nightlash-hurt-too-high-spike--r1065/

RidingKeys committed:
Fixes daze absorb and duplicate target flags

Fixes Daze effect applying on shielded tagets that absorb the damage. Also fixes a target type that had a broken function that was identical to another(Fixes Nightlash)

MadMaxMangos commented:
:white_check_mark: [Build Mangos ZERO Server 21.14.1048 completed](https://ci.appveyor.com/project/MaNGOS/server-9fytl/build/21.14.1048) (commit https://github.com/mangoszero/server/commit/6bba01270a by @RidingKeys)

RidingKeys committed:
Fix chests showing out of range even when in range

Adds a check to the new bool added IsLockInRange to make sure chests dont get hit with Spell_failed_out_of_range string when you are inside the bounding radius.

RidingKeys commented:
Added in another commit to fix this bug.
https://wowcitadel.com/forums/topic/interact-with-objects/
Tested on our server and we can confirm now that this was in fact the issue and is now 100% working.

H0zen commented on src/game/Object/Unit.cpp:
```diff
@@ -1767,28 +1767,43 @@ void Unit::DealMeleeDamage(CalcDamageInfo* damageInfo, bool durabilityLoss)
+		float Probability = 20.0f;
+
+		// Probability is 0 if they absorb
+		if (damageInfo->absorb)
```
Better add this check to line 1770 in form of negation. You don't need Probability in this case,

RidingKeys replied:
Thanks. Give me a few ill look at fixing it.

RidingKeys replied:
Changed it, gonna just double check it still works on the server and then revert the old commit and fix it to new.

MadMaxMangos commented:
:white_check_mark: [Build Mangos ZERO Server 21.14.1049 completed](https://ci.appveyor.com/project/MaNGOS/server-9fytl/build/21.14.1049) (commit https://github.com/mangoszero/server/commit/099008f551 by @RidingKeys)

RidingKeys committed:
Revert "Fix chests showing out of range even when in range"

This reverts commit eee0100baaf0bfb0ded72f9ca1b392aebdd39829.

RidingKeys committed:
Fix gameobject Out of range bug

Fix's gameobject Out of Range bug for all gameobjects that have lockId's except traps. This means that you will once again be able to use gameobjects when the cogwheel turns gold.

RidingKeys committed:
Revert "Fixes daze absorb and duplicate target flags"

This reverts commit a4f2858f34abf508853917c9d9dfeb3344e6a4c8.

RidingKeys committed:
Revert "Fix gameobject Out of range bug"

This reverts commit eab4895185d5a2266ce9fd91b93a1347b903fdce.

RidingKeys committed:
Fix Gameobjects / Fix Nightlash(Remove wrong unit chain)/ Fix Daze through shield bug

Fixes all this stuff and now has correct code to cover everything

MadMaxMangos commented:
:white_check_mark: [Build Mangos ZERO Server 21.14.1050 completed](https://ci.appveyor.com/project/MaNGOS/server-9fytl/build/21.14.1050) (commit https://github.com/mangoszero/server/commit/2b0ffadb48 by @RidingKeys)

MadMaxMangos commented:
:white_check_mark: [Build Mangos ZERO Server 21.14.1051 completed](https://ci.appveyor.com/project/MaNGOS/server-9fytl/build/21.14.1051) (commit https://github.com/mangoszero/server/commit/c2fd4e4b0f by @RidingKeys)
